packages/dataviews/src/bulk-actions-toolbar.tsx

```diff
@@ -53,58 +81,49 @@ function ActionTrigger( { action, onClick, isBusy } ) {
 	);
 }
 
-const EMPTY_ARRAY = [];
+const EMPTY_ARRAY: [] = [];
 
-function ActionButton( {
+function ActionButton< Item extends AnyItem >( {
 	action,
 	selectedItems,
 	actionInProgress,
 	setActionInProgress,
-} ) {
+}: ActionButtonProps< Item > ) {
 	const selectedEligibleItems = useMemo( () => {
 		return selectedItems.filter( ( item ) => {
-			return action.isEligible( item );
+			return ! action.isEligible || action.isEligible( item );
 		} );
 	}, [ action, selectedItems ] );
-	if ( !! action.RenderModal ) {
+	if ( 'RenderModal' in action ) {
 		return (
 			<ActionWithModal
 				key={ action.id }
 				action={ action }
 				items={ selectedEligibleItems }
 				ActionTrigger={ ActionTrigger }
-				onActionStart={ () => {
-					setActionInProgress( action.id );
-				} }
-				onActionPerformed={ () => {
-					setActionInProgress( null );
-				} }
```

Comment:
I found that these were useless.

Reply:
but these props don't seem present on the component?

Reply:
The props are present on the render modal:
https://github.com/WordPress/gutenberg/blob/4cf745843f10e2e14c822316dfe32a9161ba1b5f/packages/editor/src/components/post-actions/actions.js#L60

Reply:
Sure but then the props need to be added to `ActionWithModal` if we really want this callback here. I removed the passing of the props, not the props themselves, I didn't touch the component personally.